src/utils/azure.ts

```diff
@@ -72,30 +73,31 @@ export async function getStorageConnectionString(context: IStorageAccountWizardC
 
 export async function getEventHubsConnectionString(context: IEventHubsConnectionWizardContext & ISubscriptionContext): Promise<IResourceResult> {
     const client: EventHubManagementClient = await createEventHubClient(context);
-    const resourceGroupName: string = getResourceGroupFromId(nonNullValue(context.eventHubsNamespace?.id));
+    const rgName: string = getResourceGroupFromId(nonNullValue(context.eventHubsNamespace?.id));
     const namespaceName: string = nonNullValue(context.eventHubsNamespace?.name);
 
-    const authRulesIterable = client.namespaces.listAuthorizationRules(resourceGroupName, namespaceName);
+    const authRulesIterable = client.namespaces.listAuthorizationRules(rgName, namespaceName);
     const authRules: AuthorizationRule[] = await uiUtils.listAllIterator(authRulesIterable);
+    const manageAccessRules: AuthorizationRule[] = authRules.filter(authRule => authRule.rights?.includes('Manage'));
 
-    let authRule: string;
-    if (!authRules.length) {
-        throw new Error(localize('noAuthRules', 'Unable to locate a shared access policy for your event hub namespace.'));
-    } else if (authRules.length === 1) {
-        authRule = nonNullProp(authRules[0], 'name');
+    let authRuleName: string;
+    if (!manageAccessRules.length) {
+        authRuleName = await createAndGetAuthRuleName(context);
+    } else if (manageAccessRules.length === 1) {
+        authRuleName = nonNullProp(authRules[0], 'name');
     } else {
         const rootKeyName: string = 'RootManageSharedAccessKey';
         const placeHolder: string = localize('chooseSharedAccessPolicy', 'Choose a shared access policy.');
 
-        authRule = (await context.ui.showQuickPick(authRules.map(authRule => {
-            return { label: nonNullProp(authRule, 'name'), description: authRule.name === rootKeyName ? localize('default', '(Default)') : '' };
+        authRuleName = (await context.ui.showQuickPick(manageAccessRules.map(authRule => {
+            return { label: nonNullProp(authRule, 'name'), suppressPersistence: true, description: authRule.name === rootKeyName ? localize('default', '(default)') : '' };
```

Comment:
Why did lower-case default? Haha I prefer it with the uppercase. That's also how we do it for the (Preview) label.